### PR TITLE
Chat AntiSpam

### PR DESCRIPTION
**Pull request by blockparole**
This adds a basic regex based antispam module.

### CODE REVIEW

**Comment by blockparole** [comment]
> 
> 
> > Januar

huh?

File: src/main/java/me/zeroeightsix/kami/module/modules/misc/AntiSpam.java [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+
+        private static final String[] GREETER =
+                {
+                        // WWE
+                        "Bye, Bye .+",
+                        "Farwell, .+",
+                        // incomplete
+                };
+
+        private static final String[] DISCORD =
```
Comment (S-B99) [inline src/main/java/me/zeroeightsix/kami/module/modules/misc/AntiSpam.java]: Consider adding 
```java
                        "discord.gg",
                        "discordapp.com",
                        "discord.io",
                        "invite.dd",
``` 
to the list

Comment (S-B99) [reply src/main/java/me/zeroeightsix/kami/module/modules/misc/AntiSpam.java]: Sorry, .gg not .dd, edited

Comment (blockparole) [reply src/main/java/me/zeroeightsix/kami/module/modules/misc/AntiSpam.java]: added

**Comment by S-B99** [comment]
Also this closes #20

**Comment by blockparole** [comment]
okay, is fixed

**Comment by zeroeightysix** [comment]
Not sure how I feel about this. I'm fine with a module to prevent spam, but I don't think KAMI should have a fixed, limited set of messages to ignore.

* There's quite a lot of entries here. How are we *sure* they're spam and not an accidental match
* Should users who play on servers without these specific users who spam be left in the dark as to what the various options do?
  * Maybe there's someone spamming in chat using messages that would be blocked by a generic spam filter, but not the super-specific filter implemented here. The user wouldn't have a clue as to why their AntiSpam module isn't working.

This is difficult to solve though. I'd start by having the messages editable, preferably through a file or commands. And maybe add a little more intelligence to the filter - players who spam the same message `n` times (within a threshold) within a certain time window should be (temporarily) ignored.
Of course this also raises the question of whether or not a chat message originated from a player or not. We wouldn't want to be blocking TPA requests or the like.

**Comment by S-B99** [comment]
If you can make it read from a editable file then that solves https://github.com/S-B99/KAMI/issues/11

**Comment by zeroeightysix** [comment]
Also merging this because it received the OK from dominika.

**Pull request merged**